File: Reservations-app/src/app/models/CarReservation.ts
import { Car } from "./car.model";
import { CarCompany } from "./CarCompany.model";

export class CarReservation {
  constructor(
    public car: Car,
    public carCompany: CarCompany,
    public startDate: string,
    public endDate: string,
    public carId: number,
    public fullPrice: number,
    public userEmail?: string,
    public id?: number
  ) {}
}

/*
Ovako ce izgledati CarReservation:
    ublic id: number,
    public reservedCarId: number,
    public startDate: string,
    public endDate: string,
    public price: number,
    public userEmail: string

*/

File: ReservationAPI/ReservationAPI/Models/Rent-a-Car/CarReservation.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace ReservationAPI.Models.Rent_a_Car
{
    public class CarReservation
    {
        [Key]
        public long Id { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public double FullPrice { get; set; }
        public long CarId { get; set; }
        public virtual Car Car { get; set; }
        public string UserEmail { get; set; }
    }
}
